### PR TITLE
InteractiveTable: Add pagination and header tooltips

### PR DESCRIPTION
**Pull request by abannachGrafana**
**What is this feature?**

- `<InteractiveTable />` component.
    - add pagination
    - add ability to show an icon and tooltip in the header

**Why do we need this feature?**

- Pagination
    - Currently to paginate the `<InteractiveTable />` you need to slice your data into your page size chunks an implement your own `<Pagination />` controls. Doing so breaks the ability to sort on any column because it will only sort the subset of rows you provide.
- Header tooltips
    - If the data in a column requires description of the value you may want to use a tooltip. With a lot of rows this can quickly overwhelm the page with icons. By allowing the addition of a tooltip in the header of the column you can efficiently describe the column's values without overwhelming the user.

**Who is this feature for?**

- Users of the `<InteractiveTable />` component

**Which issue(s) does this PR fix?**:

- resolves: https://github.com/grafana/grafana-adaptive-metrics-app/issues/53

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: packages/grafana-ui/src/components/InteractiveTable/InteractiveTable.mdx [hunk 1]
```diff
@@ -164,3 +164,109 @@ export const MyComponent = () => {
   return <InteractiveTable columns={columns} data={tableData} getRowId={(r) => r.datasource} />;
 };
 ```
+
+### With pagination
+
+The table can be rendered with pagination controls by passing in the `pageSize` property.
+
+<Story id="experimental-interactivetable--with-pagination" />
```
Comment (mckn) [inline packages/grafana-ui/src/components/InteractiveTable/InteractiveTable.mdx:172]: Q: is it worth highlighting that we only support client side pagination?

**Review by mckn — APPROVED** [review]
LGTM! Added a question regarding documentation but I think the code looks good.

File: packages/grafana-ui/src/components/InteractiveTable/InteractiveTable.mdx (hunk 1 above)
Comment (abannachGrafana) [reply packages/grafana-ui/src/components/InteractiveTable/InteractiveTable.mdx]: Good idea.

File: packages/grafana-ui/src/components/InteractiveTable/InteractiveTable.mdx [hunk 2]
```diff
@@ -164,3 +164,110 @@ export const MyComponent = () => {
   return <InteractiveTable columns={columns} data={tableData} getRowId={(r) => r.datasource} />;
 };
 ```
+
+### With pagination
+
+The table can be rendered with pagination controls by passing in the `pageSize` property. All data must be provided as
```
Comment (abannachGrafana) [inline packages/grafana-ui/src/components/InteractiveTable/InteractiveTable.mdx:170]: @mckn Here is the updated doc

Comment (mckn) [reply packages/grafana-ui/src/components/InteractiveTable/InteractiveTable.mdx:170]: LGTM! 🙌🏻

**Comment by mckn** [comment]
> @mckn Would you please merge this PR? I don't have permission. What would it take for one to get permission to merge?

That is weird, you should have it via the groups you are in. Reach out to your manager and see what they say.

**Pull request merged**